### PR TITLE
[MKC-1240]Rework I2C article

### PR DESCRIPTION
**Pull request by jacobhylen**
## What This PR Changes
- This PR shifts around some content regarding I2C, by moving an information dense section from an UNO R4 Tutorial to the generic I2C guide

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/learn/05.communication/01.wire/wire.md [hunk 1]
```diff
@@ -2,16 +2,34 @@
 title: 'A Guide to Arduino & the I2C Protocol (Two Wire)'
```
Comment (jacobhylen) [inline content/learn/05.communication/01.wire/wire.md:2]:
```suggestion
title: 'Arduino & the I2C Protocol Guide'
```

Comment (jacobhylen) [reply content/learn/05.communication/01.wire/wire.md:2]: @karlsoderby Part of me thinks the title is too long as it is right now, and that this change would be better for SEO. But I'm not sure, what do you think?

Comment (karlsoderby) [reply content/learn/05.communication/01.wire/wire.md:2]: Hmm I think we can rename this simply to `Inter-Integrated Circuit (I2C) Protocol`.

File: content/learn/05.communication/01.wire/wire.md [hunk 2]
```diff
@@ -1,30 +1,92 @@
...
+- 8 bits of data
+- Another acknowledgement bit
+- 8 bits of data
+- Another acknowledgement bit
+
+But how does the controller and peripherals know where the address, messages, and so on starts and ends? That's what the SCL wire is for. It synchronises the clock of the controller with the devices, ensuring that they all move to the next instruction at the same time.
+
+However, you are nearly never going to *actually* need to consider any of this, in the Arduino ecosystem we have the [Wire library](https://www.arduino.cc/reference/en/language/functions/communication/wire/) that handles everything for you.
+
+## Qwiic & STEMMA QT
```
Comment (karlsoderby) [inline content/learn/05.communication/01.wire/wire.md:46]: I think we need to rename this section to **Hardware Connection**, and list out all available options. For example, we don't list out the most basic option which is just two wires (SCL/SDA), and we don't list Groove as well. I think the structure could be simple:

```
## Hardware Connection
There are a number of way to connect I2C devices etc etc...
### Breakout Board (Pins)
circuit + how to connect to arduino

### Groove
circuit + how to connect to arduino

### Qwiic
circuit + board used

### Stemma
circuit + how to connect to arduino
```

File: content/learn/05.communication/01.wire/wire.md [hunk 3]
```diff
@@ -1,30 +1,92 @@
...
+- `available()` - returns the number of bytes available for retrieval
+- `read()` - Reads a byte that was transmitted from a peripheral to a controller.
+- `setClock()` - Modify the clock frequency
+- `onReceive()` - Register a function to be called when a peripheral receives a transmission
+- `onRequest()` - Register a function to be called hwen a controller requests data
+- `setWireTimeout()` - Sets the timeout for transmissions in controller mode
+- `clearWireTimeoutFlag()` - Clears the timeout flag
+- `getWireTimeoutFlag()` - Checks whether a timeout has occurred since the last time the flag was cleared.
+
+## Examples
```
Comment (karlsoderby) [inline content/learn/05.communication/01.wire/wire.md:80]: So there's no examples in this section, just circuits. I think we can move this to a new section further up, and call it "Arduino I2C Pins", and just provide these circuits.

File: content/learn/05.communication/01.wire/wire.md [hunk 4]
```diff
@@ -1,30 +1,92 @@
...
 
 ***Please note that the I2C bus is attached to different pins depending on the board you are using. For example, the pins used for [MKR WiFi 1010](/hardware/mkr-wifi-1010) are D11, D12, while the pins for [UNO](/hardware/uno-rev3) are D18, D19. See the image below to understand how to locate the correct pins on your board.***
 
 ![I2C pins on the UNO.](assets/uno-i2c.png)
 
 ![I2C pins on the MKR WiFi 1010](assets/wifi-1010-i2c.png)
 
-## Tutorials
-
+## Tutorials
```
Comment (karlsoderby) [inline content/learn/05.communication/01.wire/wire.md:89]: We should spend some time updating this section, maybe we can combine it with what I proposed in the "Stemma/Qwiic" section, and provide some newer examples for popular breakouts. This is outside the original scope so let's talk more about this next week.

File: content/learn/05.communication/01.wire/wire.md [hunk 5]
```diff
@@ -1,30 +1,92 @@
 ---
-title: 'A Guide to Arduino & the I2C Protocol (Two Wire)'
+title: 'Inter-Integrated Circuit (I2C) Protocol'
 description: 'Allows the communication between devices or sensors connected via Two Wire Interface Bus.'
 tags: [I2C, Wire, SDA, SCL]
```
Comment (karlsoderby) [inline content/learn/05.communication/01.wire/wire.md:4]:
```suggestion
tags: [I2C, Wire, SDA, SCL, Qwiic, Groove]
```

File: content/learn/05.communication/01.wire/wire.md [hunk 6]
```diff
@@ -1,30 +1,92 @@
...
-author: 'Arduino, Nicholas Zambetti'
+author: 'Nicholas Zambetti, Karl Söderby, Jacob Hylén'
 difficulty: 'intermediate'
 ---
+## Introduction
```
Comment (karlsoderby) [inline content/learn/05.communication/01.wire/wire.md:8]: We should state somewhere that all Arduino boards support the I2C protocol.

**Review by karlsoderby — CHANGES_REQUESTED** [review]
Looks great, but left some comments that we can work on together next week. I think we can re-organize some stuff.

We might also want to add a list of all boards with their corresponding pins (for each port, like the DUE/MEGA/GIGA has more than 1).

**Review by karlsoderby — APPROVED** [review]

**Pull request merged**